operator/bucketcontroller/delete.go

```diff
@@ -55,12 +56,28 @@ func (p *ProvisioningPipeline) deleteAllObjects(ctx *pipelineContext) error {
 		}
 	}()
 
-	for obj := range p.minioClient.RemoveObjects(ctx, bucketName, objectsCh, minio.RemoveObjectsOptions{GovernanceBypass: true}) {
+	bypassGovernance, err := p.isBucketLockEnabled(ctx, bucketName)
+	if err != nil {
+		return fmt.Errorf("cannot determine bucketLockConfig of bucket %s: %w", bucketName, err)
```

Comment:
Nitpick: I'd probably suggest that in case of an error, we log the error, assume that there is no lock and try to delete the objects anyway, instead of aborting.

Mainly because of the brittle error matching below, but also if there maybe are other errors that would still allow us to delete the objects.

Reply:
Fine by me, if the deletion of the objects fails we'll still see it as an event then.